test/cri-containerd/container_update_test.go

```diff
@@ -0,0 +1,334 @@
+		runtimeHandler   string
+		sandboxImage     string
+		containerImage   string
+		cmd              []string
+	}
+	tests := []config{
+		{
+			name:             "WCOW_Process",
+			requiredFeatures: []string{featureWCOWProcess},
+			runtimeHandler:   wcowProcessRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "WCOW_Hypervisor",
+			requiredFeatures: []string{featureWCOWHypervisor},
+			runtimeHandler:   wcowHypervisorRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "LCOW",
+			requiredFeatures: []string{featureLCOW},
+			runtimeHandler:   lcowRuntimeHandler,
+			sandboxImage:     imageLcowK8sPause,
+			containerImage:   imageLcowAlpine,
+			cmd:              []string{"top"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, []string{test.sandboxImage})
+			} else if test.runtimeHandler == wcowHypervisorRuntimeHandler {
+				pullRequiredImages(t, []string{test.sandboxImage})
+			}
+
+			podRequest := &runtime.RunPodSandboxRequest{
+				Config: &runtime.PodSandboxConfig{
+					Metadata: &runtime.PodSandboxMetadata{
+						Name:      t.Name(),
+						Namespace: testNamespace,
+					},
+				},
+				RuntimeHandler: test.runtimeHandler,
+			}
+
+			client := newTestRuntimeClient(t)
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			podID := runPodSandbox(t, client, ctx, podRequest)
+			defer removePodSandbox(t, client, ctx, podID)
+			defer stopPodSandbox(t, client, ctx, podID)
+
+			containerRequest := &runtime.CreateContainerRequest{
+				Config: &runtime.ContainerConfig{
+					Metadata: &runtime.ContainerMetadata{
+						Name: t.Name() + "-Container",
+					},
+					Image: &runtime.ImageSpec{
+						Image: test.containerImage,
+					},
+					Command: test.cmd,
+				},
+				PodSandboxId:  podID,
+				SandboxConfig: podRequest.Config,
+			}
+
+			containerID := createContainer(t, client, ctx, containerRequest)
+			defer removeContainer(t, client, ctx, containerID)
+
+			startContainer(t, client, ctx, containerID)
+			defer stopContainer(t, client, ctx, containerID)
+
+			// make request to increase cpu shares
+			updateReq := &runtime.UpdateContainerResourcesRequest{
+				ContainerId: containerID,
+			}
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				updateReq.Linux = &runtime.LinuxContainerResources{
+					CpuShares: 5000,
+				}
+			} else {
+				updateReq.Windows = &runtime.WindowsContainerResources{
+					CpuShares: 5000,
+				}
+			}
+
+			if _, err := client.UpdateContainerResources(ctx, updateReq); err != nil {
```

Comment:
In this and rest of the tests can we do some verification (with a `stats` call or something) if the updated resource value (i.e the cpu count or memory size) matches our expectations?

Reply:
only issue here is that the stats calls don't actually return info on the limits, just what has been used. If we want this, we'll need to use my test utility tool, which I'm fine doing.

Reply:
Now that your other PR for shimdiag exec is merged,  can we add this verification here?

Reply:
@ambarve how do you feel about a separate PR with the tests in them? Or a separate commit? There's a lot of code for it 😅